### PR TITLE
Handle time jumping even if the user clicks above/below the slider

### LINKED ISSUE #171: Video Time Navigation Doesn't Change When Clicking Above/Below Slider
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug
If you click above/below the progress slider but still within the bottom bar, the video doesn't navigate there despite the preview popup is showing. This is a nuisance as it requires the user to be overly precise to click the slider itself. I would expect the player to be forgiving and let the user select a time on the slider wherever the preview popup shows.

Below is a recording demonstrating the current behavior. You'll see the widget that is showing my mouse clicks.

https://imgur.com/RbeIs67

### To Reproduce
1. Open a video (can be paused or playing)
1. Press below the slider (as shown in the above recording). Pressing above will sometimes work depending on how far away from the slider you press
1. Notice the video doesn't change the progress/time

### Expected behavior
Pressing anywhere above or below the time progress bar within the bottom bar should change the time.

## Platform Information
image

- [x] I'm using the latest version from git that I've manually compiled
- [x] I'm using the latest released stable version

### PR DESCRIPTION
**Pull request by seesemichaelj**
This PR fixes #171 by setting the slider value manually given the click event rather than relying on the Slider class to properly change the value.

Current behavior: https://imgur.com/RbeIs67
New behavior: https://imgur.com/3GWXq8I

### CODE REVIEW

**Comment by jeremypw** [comment]
Is it possible to increase the height of the slider track instead? It feels a bit wrong to interfere with the native click handling.  A drawback of your approach is that I suspect the scale value will be set twice if the user clicks on the track, which could cause a glitch if there is a delay between press and release.

**Comment by jeremypw** [comment]
Also you would not be able to drag the slider if clicking outside the track.

**Comment by seesemichaelj** [comment]
>Is it possible to increase the height of the slider track instead?

Sure this is a valid solution, but it would require a design change that would postpone a fix for this issue. Who knows how long that could take, when this is a valid solution today. I could see a potential solution to increase the clickable region (looks like #174 might implement this? going to test your PR) for the slider but not change the visual appearance as another quick fix. I have a feeling we'll get a bit of push back to make the slider wider.

>It feels a bit wrong to interfere with the native click handling. 

Ya I get where you're coming from, but it's not uncommon for there to be an increased margin for small elements so that they're easier to click. Frankly, it feels a bit wrong to be using a slider for the `TimeWidget` to begin with; I would have made a custom element for this functionality. This PR just deals with the fact we're using the wrong tool for the job to start with with as minimal changes as possible.

>A drawback of your approach is that I suspect the scale value will be set twice if the user clicks on the track, which could cause a glitch if there is a delay between press and release.

I'd encourage you to test this; I verified this doesn't happen. I also tested other edge cases and believe there aren't any regressions in this fix.

>Also you would not be able to drag the slider if clicking outside the track.

This is correct; but you weren't able to do this before so this PR isn't a regression.

**Comment by jeremypw** [comment]
The comment about dragging was drawing attention to the fact that this introduces a clickable but not draggable area whereas the native behaviour is the clickable area is all draggable.  It may not be a blocker though.

You are right that setting the scale value twice is not actually a problem because the video does not actually navigate until button release even though the scale value changes on both button press and button release. It is possible to demonstrate by pressing off trough (changes value)  move pointer (does not drag) and release (value changes again).

Unless I can overcome the drawbacks with the css PR you identified this may prove to be the preferred approach.

File: src/Widgets/Player/TimeWidget.vala [hunk 1]
```diff
@@ -59,6 +59,14 @@ public class Audience.Widgets.TimeWidget : Granite.SeekBar {
         });
 
         scale.button_release_event.connect ((event) => {
```
Comment (jeremypw) [inline src/Widgets/Player/TimeWidget.vala]: Try linking to the seekbar button release event instead - it seems to get rid of the small remaining region where the popover may be showing but the value does not change on release (best seen using window_scale >=2) and I have not noticed any regressions.

**Comment by seesemichaelj** [comment]
Ok sounds good! I'll get to this in a couple days as I'm traveling

File: src/Widgets/Player/TimeWidget.vala (hunk 1 above)
Comment (seesemichaelj) [reply src/Widgets/Player/TimeWidget.vala]: Sorry for the delay. Things in production at work have been chaotic. I changed `scale` to `this` here and couldn't see a difference. I'm guessing it's because I'm not sure how to set `window_scale`. However, since I didn't notice a difference in my environment, I went ahead and committed the change. Let me know if this works better for you and if there's anything else

Comment (jeremypw) [reply src/Widgets/Player/TimeWidget.vala]: I was using `GTK_DEBUG=interactive io.elementary.videos` to test at different window scales.
